### PR TITLE
BookA-Pset_SpatialGeoObsSample

### PR DESCRIPTION
**Pull request by larswik**
# Pset: Pset_SpatialGeoObsSample

## Description: 

Common properties for samples

| Property | Datatype | Description |
|----------|----------|-------------|
|SampleType|IfcLabel|Type of a sample with respect to sampling method and material, with e.g. enumerations as defined in AGS  |
|SampleBoreholeDepthRange|IfcNonNegativeLengthMeasure|Relative top and bottom depths of a borehole where a sample was retrieved|
|Identifier|IfcIdentifier|Identifier of a sample like No. and name.|
|SampleProcedure|IfcText|Description of steps performed by a Sampler in order to extract a Sample from its sampledFeature in the frame of a Sampling.|
|SamplePurpose|IfcLabel|Provides the reason why the sample was collected, planed methods for testingand analysis|
|SamplingPointType|IfcLabel|The sourceLocation is the location from where the MaterialSample was obtained, e.g. outcrop point, tunnel chainage, any other local reference|

### CODE REVIEW

File: IFC4x3/Properties/i/Identifier_3CVqGHgfj9xui_$tglS6m2/DocProperty.xml
Comment (SergejMuhic) [inline IFC4x3/Properties/i/Identifier_3CVqGHgfj9xui_$tglS6m2/DocProperty.xml]: This seems more like `IfcRoot.GlobalId` i. e. the sample id of `IfcGeoScienceElement` and not the material.

File: IFC4x3/Properties/s/SampleBoreholeDepthRange_16N06qRHf0NOknS38zpWxi/DocProperty.xml
Comment (SergejMuhic) [inline IFC4x3/Properties/s/SampleBoreholeDepthRange_16N06qRHf0NOknS38zpWxi/DocProperty.xml]: `IfcPropertyBoundedValue`?

File: IFC4x3/Properties/s/SamplingPointType_3snE6alTz82vidqo5JBy_l/Documentation.md
Comment (SergejMuhic) [inline IFC4x3/Properties/s/SamplingPointType_3snE6alTz82vidqo5JBy_l/Documentation.md]: What is "sourceLocation"? Any reference to its definition?

Also, the description seems to not match to the property name.

File: IFC4x3/Properties/i/Identifier_3CVqGHgfj9xui_$tglS6m2/DocProperty.xml
Comment (larswik) [reply IFC4x3/Properties/i/Identifier_3CVqGHgfj9xui_$tglS6m2/DocProperty.xml]: I don't think GlobalId covers what is intended here. The identifier feels to me more "man made" such as in IfcRoot.Name. In this case the structure was intended as follows:
IfcGeoScienceObservation/LABTEST or INSITUTEST => IfcRelAssociatesMaterial => IfcMaterialDefinition.HasProperties => Pset_SpatialGeoObsSample. However, after looking at the properties, I agree with @SergejMuhic and would like to simplify the structure to: IfcGeoScienceObservation/LABTEST or INSITUTEST => IfcRelDefinesByProperties => Pset_SpatialGeoObsSample since the Pset seems to have more to do with the sampling rather than the sample.

Proposal:
- Applicability : IfcGeoScienceObservation/LABTEST or INSITUTEST
- Remove Identifier property and instead use IfcGeoScienceObservation.Name

Ping @j-weil.

File: IFC4x3/Properties/s/SampleBoreholeDepthRange_16N06qRHf0NOknS38zpWxi/DocProperty.xml
Comment (larswik) [reply IFC4x3/Properties/s/SampleBoreholeDepthRange_16N06qRHf0NOknS38zpWxi/DocProperty.xml]: > `IfcPropertyBoundedValue`?

Yes, it should be a bounded value (it says so also in the excel).

File: IFC4x3/Properties/s/SamplingPointType_3snE6alTz82vidqo5JBy_l/Documentation.md
Comment (larswik) [reply IFC4x3/Properties/s/SamplingPointType_3snE6alTz82vidqo5JBy_l/Documentation.md]: I think that this needs a clarification. Ping @j-weil (sorry).

File: IFC4x3/Properties/s/SampleBoreholeDepthRange_16N06qRHf0NOknS38zpWxi/DocProperty.xml
Comment (larswik) [reply IFC4x3/Properties/s/SampleBoreholeDepthRange_16N06qRHf0NOknS38zpWxi/DocProperty.xml]: Fixed in [eca1bf0](https://github.com/bSI-InfraRoom/IFC-Specification/pull/592/commits/eca1bf08b4def82f2ef81f464d3d863c04ef5687)

File: IFC4x3/Properties/s/SamplingPointType_3snE6alTz82vidqo5JBy_l/Documentation.md
Comment (j-weil) [reply IFC4x3/Properties/s/SamplingPointType_3snE6alTz82vidqo5JBy_l/Documentation.md]: This refers to a "label" kind of description to respect local or project specific location names or IDs, such as "key ourcrop numer xxx", a building, a sampling ID, anything that migh not be covered by other predefined objects ir lists in the model

Comment (larswik) [reply IFC4x3/Properties/s/SamplingPointType_3snE6alTz82vidqo5JBy_l/Documentation.md]: Ok, then I would propose re-writing the definition to something like: "Specifies the location from where the material sample was obtained, e.g. outcrop point, tunnel chainage, any other local reference."

Comment (SergejMuhic) [reply IFC4x3/Properties/s/SamplingPointType_3snE6alTz82vidqo5JBy_l/Documentation.md]: This sounds good to me. Will wait for approval before merging.

**Review by SergejMuhic — APPROVED** [review]

File: IFC4x3/Properties/i/Identifier_3CVqGHgfj9xui_$tglS6m2/DocProperty.xml
Comment (larswik) [reply IFC4x3/Properties/i/Identifier_3CVqGHgfj9xui_$tglS6m2/DocProperty.xml]: Done.

File: IFC4x3/Properties/s/SamplingPointType_3snE6alTz82vidqo5JBy_l/Documentation.md
Comment (larswik) [reply IFC4x3/Properties/s/SamplingPointType_3snE6alTz82vidqo5JBy_l/Documentation.md]: Done.

File: IFC4x3/Properties/s/SamplingPointType_3snE6alTz82vidqo5JBy_l/DocProperty.xml
Comment (SergejMuhic) [inline IFC4x3/Properties/s/SamplingPointType_3snE6alTz82vidqo5JBy_l/DocProperty.xml]: What happened here? Manual editing? 😄 
```suggestion
<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="SamplingPointType_3snE6alTz82vidqo5JBy_l" Name="SamplingPointType" UniqueId="f6c4e1a4-bddf-480b-9b27-d321532fcfaf" PrimaryDataType="IfcLabel" />
```

**Review by SergejMuhic — APPROVED** [review]

**Pull request merged**
